docs/app/js/codepen.js

```diff
@@ -0,0 +1,175 @@
+     * @param {Array} externalScripts Any additional dependent scripts
+     * @usage
+     * var demo = {
+     *   id: 'demo-id',
+     *   title: 'title',
+     *   module: 'module-name',
+     *   files: {
+     *     html: [{ name: 'relative-path', contents: '...'}],
+     *     css: [{ name: 'relative-path', contents: '...'],
+     *     index: { contents: '...' },
+     *     js: [{ contents: '...' }]
+     *    }
+     * };
+     *
+     * var externalScripts = ['pathToScript1', 'pathToScript2'];
+     *
+     * $codepen.editOnCodepen(demo, externalScripts);
+     */
+    function translate(demo, externalScripts) {
+      var files = demo.files;
+
+      return {
+        title: demo.title,
+        html: processHtml(demo),
+        css: mergeFiles(files.css).join(' '),
+        js: processJs(files.js),
+        js_external: externalScripts.concat([CORE_JS, ASSET_CACHE_JS]).join(';'),
+        css_external: CORE_CSS
+      };
+    };
+
+    function processHtml(demo) {
+      var index = demo.files.index.contents;
+
+      var processors = [
+        applyAngularAttributesToParentElement,
+        insertTemplatesAsScriptTags,
+        replaceEscapedCharacters
+      ];
+
+      angular.forEach(processors, function(processor) {
+        index = processor(index, demo);
+      });
+
+      return index;
+    };
+
+    function processJs(jsFiles) {
+      var mergedJs = mergeFiles(jsFiles).join(' ');
+      var script = replaceDemoModuleWithCodepenModule(mergedJs);
+      return script;
+    };
+
+    function mergeFiles(files) {
+      return files.map(function(file) {
+        return file.contents;
+      });
+    };
+
+    //Adds class to parent element so that styles are applied correctly
+    //Adds ng-app attribute.  This is the same module name provided in the asset-cache.js
+    function applyAngularAttributesToParentElement(html, demo) {
+      var tmp = angular.element(html);
+      tmp.addClass(demo.id);
+      tmp.attr('ng-app', 'MyApp');
+      return tmp[0].outerHTML;
+    };
+
+    //Adds templates inline in the html, so that templates are cache in the example
+    function insertTemplatesAsScriptTags(indexHtml, demo) {
+      if (demo.files.html.length) {
+        var tmp = angular.element(indexHtml);
+        angular.forEach(demo.files.html, function(template) {
+          tmp.append("<script type='text/ng-template' id='" + template.name + "'>" + template.contents + "</script>");
+        });
+        return tmp[0].outerHTML;
+      }
+      return indexHtml;
+    };
+
+    //Escapes ampersands so that after codepen unescapes the html the escaped code block
+    //uses the correct escaped characters
+    function replaceEscapedCharacters(html) {
+      return html
+        .replace(/&gt;/g, "&amp;gt;")
+        .replace(/&lt;/g, "&amp;lt;");
+    };
+
+    //Required to make codepen work.  Without replacing the module, the asset cache will not work
```

Comment:
Expand comment with more details, e.g., _why_ asset cache won't work without replacing the module name.